zebra/interface.c

```diff
@@ -513,13 +513,14 @@ void if_add_update(struct interface *ifp)
 {
 	struct zebra_if *if_data;
 	struct zebra_ns *zns;
+	struct zebra_vrf *zvrf = vrf_info_lookup(ifp->vrf_id);
 
-	if (vrf_is_backend_netns())
-		zns = zebra_ns_lookup((ns_id_t)ifp->vrf_id);
+	/* case interface populate before vrf enabled */
+	if (zvrf->zns)
+		zns = zvrf->zns;
```

Comment:
Why would we not have a zns pointer at this point in time?  The whole point of the abstraction is *to* not have to have these if's all throughout the code base

Reply:
This part of code is specific. we are in the case where a VRF has been created, but VRF is not yet enabled (  in the case of vrf-lite). So zns pointer is NULL if you don't rely on zebra_ns_lookup.

If you happen to have vrf NETNS, it is ok.
Maybe some rework in vrf-lite process should be reviewed,  I think. But I don't think it should be considered as part of what the review brings in.